src/librbd/internal.cc

```diff
@@ -2862,10 +2864,16 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
     }
 
     if (next_mirror_mode == cls::rbd::MIRROR_MODE_IMAGE) {
+      if (current_mirror_mode == cls::rbd::MIRROR_MODE_POOL) {
```

Comment:
I would actually put this and the next message within the rbd CLI prefixed with "note: " instead of "warning: " (since it isn't technically wrong).